index.js

```diff
@@ -35,6 +36,29 @@ class Child {
   }
 }
 
+class KeyCache {
+  constructor () {
+    this.keys = new Xache({ maxSize: 65536 })
+    this.length = 0
+  }
+
+  get (seq) {
+    return this.keys.get(seq) || null
+  }
+
+  set (seq, key) {
+    this.keys.set(seq, key)
+    if (seq >= this.length) this.length = seq + 1
+  }
+
+  gc (length) {
+    for (let i = length; i < this.length; i++) {
+      this.keys.delete(i)
+    }
+    this.length = length
```

Comment:
prop good have someone else think about this one.

you can use a linked list to make this faster on paper, but this is optimised for small truncations.

We should prop add some safety that if this.length - length > THRESHOLD we just this.keys.clear()